### PR TITLE
feat: Tx and Query Command Line for Redeem Data Pass

### PR DESCRIPTION
**Pull request by inchori**
- Command-Line for Redeem Data Pass.
- The key for Redeem Data Pass was changed because of the pagination.

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ const Name = "panacea"
 var (
 	// If EnabledSpecificWasmProposals is "", and this is "true", then enable all x/wasm proposals.
 	// If EnabledSpecificWasmProposals is "", and this is not "true", then disable all x/wasm proposals.
-	WasmProposalsEnabled = "false"
+	WasmProposalsEnabled = "true"
```
Comment (hansol-medi) [inline app/app.go:122]: You have to revert this change I think.

File: proto/panacea/datapool/v2/query.proto [hunk 2]
```diff
@@ -35,6 +35,11 @@ service Query {
   rpc DataValidationCertificates(QueryDataValidationCertificatesRequest) returns (QueryDataValidationCertificatesResponse) {
     option (google.api.http).get = "/panacea/datapool/v2/data_validation_certificates/pool/{pool_id}/round/{round}";
   }
+
+  // DataPassRedeemReceipts returns DataPassRedeemReceipts
+  rpc DataPassRedeemReceipts(QueryDataPassRedeemReceiptsRequest) returns (QueryDataPassRedeemReceiptsResponse) {
+    option (google.api.http).get = "/panacea/datapool/v2/data_pass_redeem_receipts/pool/{pool_id}";
+  }
```
Comment (hansol-medi) [inline proto/panacea/datapool/v2/query.proto:42]: This query returns the list of `DataPassRedeemReceipts`.
IMO, we need a query which returns a `DataPassRedeemReceipt` by token id. It may be used in [data download api](https://github.com/medibloc/panacea-data-market-validator/issues/49) of data validator to get and validate `DataPassRedeemReceipt`. 
Buyer can buy multiple `DataPass`es and redeem them, so we have to make the redeemed `DataPass` and its receipt be specific

File: x/datapool/client/cli/queryPool.go [hunk 3]
```diff
@@ -119,3 +119,47 @@ func CmdGetDataValidationCertificates() *cobra.Command {
...
+
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
```
Comment (hansol-medi) [inline x/datapool/client/cli/queryPool.go:162]: We need to add pagination flags to the cmd as well to enable to accept pagination flags.
```go
flags.AddPaginationFlagsToCmd(cmd, "all data pass")
```

File: x/datapool/keeper/grpc_query_pool.go [hunk 4]
```diff
@@ -102,3 +102,39 @@ func (k Keeper) DataValidationCertificates(goCtx context.Context, req *types.Que
...
+	redeemer, err := sdk.AccAddressFromBech32(req.Redeemer)
+	if err != nil {
+		return nil, err
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	redeemReceiptStore := prefix.NewStore(store, types.GetKeyPrefixNFTRedeemReceiptByPoolID(req.PoolId, redeemer))
+
+	var redeemReceipts []types.DataPassRedeemReceipt
+	pageRes, err := query.Paginate(redeemReceiptStore, req.Pagination, func(_ []byte, value []byte) error {
```
Comment (hansol-medi) [inline x/datapool/keeper/grpc_query_pool.go:122]:
```suggestion
	pageRes, err := query.Paginate(redeemReceiptStore, req.Pagination, func(_, value []byte) error {
```

**Review by gyuguen — APPROVED** [review]
LGTM 👍

File: proto/panacea/datapool/v2/query.proto (hunk 2 above)
Comment (inchori) [reply proto/panacea/datapool/v2/query.proto:42]: The query a `DataPassRedeemReceipt` was implemented in the branch #322 .

File: proto/panacea/datapool/v2/query.proto [hunk 5]
```diff
@@ -86,4 +91,19 @@ message QueryDataValidationCertificatesResponse {
...
   cosmos.base.query.v1beta1.PageResponse pagination = 2;
-}
\ No newline at end of file
+}
+
+// QueryDataPassRedeemReceiptsRequest is the request type for the Query/DataPassRedeemReceipts RPC method.
+message QueryDataPassRedeemReceiptsRequest {
+  uint64 pool_id = 1;
+  string redeemer = 3;
+  cosmos.base.query.v1beta1.PageRequest pagination = 4;
```
Comment (hansol-medi) [inline proto/panacea/datapool/v2/query.proto:100]: You can use second argument. It is missed when removing round I think :)
And I wonder why you remove the round.

File: x/datapool/client/cli/queryPool.go [hunk 6]
```diff
@@ -119,3 +119,47 @@ func CmdGetDataValidationCertificates() *cobra.Command {
...
+
+			if err != nil {
+				return err
+			}
+
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddPaginationFlagsToCmd(cmd, "all data pass")
```
Comment (hansol-medi) [inline x/datapool/client/cli/queryPool.go:162]: Please add query flags
```suggestion
	flags.AddPaginationFlagsToCmd(cmd, "all data pass")
	flags.AddQueryFlagsToCmd(cmd)
```

File: proto/panacea/datapool/v2/query.proto (hunk 5 above)
Comment (inchori) [reply proto/panacea/datapool/v2/query.proto:100]: This will be fixed in #322 branch.

File: x/datapool/client/cli/queryPool.go (hunk 6 above)
Comment (inchori) [reply x/datapool/client/cli/queryPool.go:162]: This will be fixed in #322 branch.

**Review by hansol-medi — APPROVED** [review]
I checked you've fixed the things in the next PR, so approved.

**Review by youngjoon-lee — COMMENTED** [review]
lgtm
